scripts/add_existing_baseyear.py

```diff
@@ -434,12 +443,23 @@ def add_heating_capacities_installed_before_baseyear(
         else:
             efficiency = costs.at[costs_name, "efficiency"]
 
-        for i, grouping_year in enumerate(grouping_years):
-            if int(grouping_year) + default_lifetime <= int(baseyear):
-                continue
+        valid_grouping_years = pd.Series(
+            [
+                int(gy)
+                for gy in grouping_years
+                if int(baseyear) - default_lifetime <= int(gy) < int(baseyear)
```

Comment:
This changes the definition of the valid grouping years, e.g.
you have `baseyear=2020`, `default_lifetime=20`, `grouping_year=gy=2000`

Then before 2000 + 20 <= 2020 -> True -> would not be considered

Now: 2020-20 <= 2000 -> True would be considered